Fixed #31275 -- Optimized sql_flush() without resetting sequence on MySQL. (#12634)
django/django: docs/releases/3.1.txt

* The new :class:`~django.db.models.F` expression ``bitxor()`` method allows
  :ref:`bitwise XOR operation <using-f-expressions-in-filters>`.

* The ``sql_flush`` method in MySQL DatabaseOperations is faster now.

> 

Reviewer: What about

```suggestion
* The command:`sql_flush` command now generates more efficient SQL for table containing
  a few rows on MySQL by using ``DELETE`` statements instead of ``TRUNCATE``.
```

I think another mention should be added to the `Tests` section

```
* :class:`django.test.TransactionTestCase` teardown is now faster on MySQL due to
  command:`sql_flush` improvements on this backend. As a side effect the latter doesn't
  automatically reset sequences on teardown anymore. Enable
  :attr:`django.test.TransactionTestCase.reset_sequences` if your tests require such feature.
```

Author: Thank you for your suggestion. I added your suggestion in `Tests` section at a52ab51638.

* :meth:`.QuerySet.bulk_create` now sets the primary key on objects when using
  MariaDB 10.5+.

> * The ``DatabaseOperations.sql_flush()`` method now generates more efficient
>   SQL for table containing a few rows on MySQL by using
>   ``DELETE`` statements instead of ``TRUNCATE``.

Author: ```suggestion
* The ``DatabaseOperations.sql_flush()`` method now generates more efficient
  SQL for table which doesn't need to reset sequences on MySQL by using
  ``DELETE`` statements instead of ``TRUNCATE``.
```
